cedar-lean/Cedar/Thm/Data/Control.lean

```diff
@@ -0,0 +1,49 @@
+/-
+ Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-/
+
+import Std
+
+/-!
+This file contains `simp` lemmas for proofs about code that uses the `do`
+notation together with `Except` and `Option`.
+-/
```

Comment:
Did we have this file the whole time?  this would have been useful to know.  I'll have to make sure I'm always importing `Thm.Data.Control` when I'm working with `do` notation

Reply:
Yes, but it was named less clearly (Thm.Core.Std) and easy to miss :-\. I've renamed it to Control to match Std naming for things to related with the `do` notation.  You definitely want to import this for any proofs that deal with `do` code; you can avoid a lot of boilerplate (see eg what the first `simp` does in the proof of Cedar.Thm.Data.List.foldlM_of_assoc)